lib/elixir/lib/module.ex

```diff
@@ -1514,6 +1514,66 @@ defmodule Module do
     end
   end
 
+  @doc """
+  Gets the last set value of a given attribute from a module.
+
+  If the attribute was marked with `accumulate` with
+  `Module.register_attribute/3`, the previous value to have been set will be
+  returned. If the attribute does not accumulate, this call is the same as
+  calling `Module.get_attribute/3`.
+
+  This function can only be used on modules that have not yet been compiled.
+  Use the `c:Module.__info__/1` callback to get all persisted attributes, or
+  `Code.fetch_docs/1` to retrieve all documentation related attributes in
+  compiled modules.
+
+  ## Examples
+
+      defmodule Foo do
+        Module.put_attribute(__MODULE__, :value, 1)
+        Module.get_last_attribute(__MODULE__, :value) #=> 1
+
+        Module.get_last_attribute(__MODULE__, :not_found, :default) #=> :default
+
+        Module.register_attribute(__MODULE__, :acc, accumulate: true)
+        Module.put_attribute(__MODULE__, :acc, 1)
+        Module.get_last_attribute(__MODULE__, :acc) #=> 1
+        Module.put_attribute(__MODULE__, :acc, 2)
+        Module.get_last_attribute(__MODULE__, :acc) #=> 2
+      end
+
+  """
+  @doc since: "1.15.0"
+  @spec get_last_attribute(module, atom, term) :: term
+  def get_last_attribute(module, key, default \\ nil) when is_atom(module) and is_atom(key) do
+    assert_not_compiled!(
+      {:get_last_attribute, 2},
+      module,
+      "Use the Module.__info__/1 callback or Code.fetch_docs/1 instead"
+    )
+
+    {set, bag} = data_tables_for(module)
+
+    case :ets.lookup(set, key) do
+      [{_, _, :accumulate, _}] ->
```

Comment:
We also need to copy the "traces" handling code from get_attribute. We can pretty much copy it as is. The commit that added traces is here: https://github.com/elixir-lang/elixir/commit/7913d5d02e44380aeb3935792affdbdd215da9a4

You can add a test like these: https://github.com/elixir-lang/elixir/commit/7913d5d02e44380aeb3935792affdbdd215da9a4#diff-e490c25a76a5cc37928b87a55ee5e43aae262c90eb1d460109cfa268ed41167cR214

The idea is that you store an attribute, call get_last_attribute in the module body, and any alias inside the attribute will be a compile-time dependency.

Reply:
Got it! I will add the traces bit over. I thought perhaps it was only for the private `Kernel.@` usage.

Reply:
Question: looking at the tests, in this scenario, would be expect both String and Enum to be tracked, or just Enum?

```elixir
defmodule Foo do
  Module.register_attribute(__MODULE__, :attr, accumulate: true)

  @attr String
  @attr Enum

  _ = Module.get_last_attribute(__MODULE__, :attr) #=> Enum
end

Reply:
If we can trace only one of them, it is great, but if not, no problem.